qtfred/src/ui/dialogs/ShipEditor/ShipEditorDialog.cpp

```diff
@@ -879,7 +879,8 @@ void ShipEditorDialog::on_restrictArrivalPathsButton_clicked()
 }
 void ShipEditorDialog::on_customWarpinButton_clicked()
 {
-	// TODO:: Custom warp Dialog
+	auto dialog = new dialogs::ShipCustomWarpDialog(this, _viewport, false);
```

Comment:
Is this going to cause a memory leak?  What happens to the `dialog` pointer when it goes out of scope?

Reply:
I did a bit of searching around.  Basically, there are a few options:
1. Display a modal dialog with `exec()` rather than `show()` and then delete the pointer immediately afterward
2. Wait until the program exits at which point the child dialogs will be deleted automatically (requires the dialog to be created with a parent)
3. Use a global variable so there is a single dialog that is hidden or shown as needed, rather than being created each time
4. Set the delete-on-close attribute via `diag->setAttribute(Qt::WA_DeleteOnClose);`

Reply:
Option 3 is what I attempted with the other dialogs and prove a bit kludge and difficult to manage making sure it had the right data. Of these I'm leaning towards option 4 being best for our use cases.

Reply:
Agreed, option 4 sounds pretty good